python/lsst/ts/standardscripts/maintel/take_aos_sequence_comcam.py

```diff
@@ -224,88 +243,97 @@ async def configure_tcs(self) -> None:
+                    timeout=self.camera.fast_timeout,
+                )
+            )
 
-        supplemented_group_id = self.next_supplemented_group_id()
+        self.log.debug("Moving to in-focus position")
+        await self.move_hexapod(0)
 
-        intra_visit_id = await self.camera.take_cwfs(
+        self.log.info("Taking in-focus image")
+        await self.camera.take_acq(
             exptime=self.exposure_time,
             n=1,
-            group_id=supplemented_group_id,
+            group_id=self.group_id,
             filter=self.filter,
-            reason="INTRA" + ("" if self.reason is None else f"_{self.reason}"),
+            reason="INFOCUS" + ("" if self.reason is None else f"_{self.reason}"),
             program=self.program,
             note=self.note,
         )
 
-        self.log.debug("Moving to extra-focal position")
-
-        # Hexapod offsets are relative, so need to move 2x the offset
-        # to get from the intra- to the extra-focal position.
-        z_offset = -(self.dz * 2.0)
-        await self.mtcs.offset_camera_hexapod(x=0, y=0, z=z_offset, u=0, v=0)
+        if self.mode == Mode.TRIPLET:
+            try:
+                await ocps_execute_task
+            except Exception:
+                self.log.exception("Executing OCPS task failed. Ignoring.")
 
-        self.log.info("Taking extra-focal image")
+    async def move_hexapod(self, target_z: float) -> None:
+        """Move the hexapod to the target z position
+        (relative to current position).
 
-        extra_visit_id = await self.camera.take_cwfs(
-            exptime=self.exposure_time,
-            n=1,
-            group_id=supplemented_group_id,
-            filter=self.filter,
-            reason="EXTRA" + ("" if self.reason is None else f"_{self.reason}"),
-            program=self.program,
-            note=self.note,
-        )
-
-        self.log.info("Send processing request to RA OCPS.")
-        config = {
-            "LSSTComCamSim-FROM-OCS_DONUTPAIR": f"{intra_visit_id[0]},{extra_visit_id[0]}"
-        }
-        ocps_execute_task = asyncio.create_task(
-            self.ocps.cmd_execute.set_start(
-                config=json.dumps(config),
-                timeout=self.camera.fast_timeout,
-            )
-        )
-
-        self.log.debug("Moving to in-focus position")
+        Parameters
+        ----------
+        target_z : `float`
+            Target z position to move to (relative offset).
+        """
+        z_move = target_z - self.current_z_position  # Calculate relative move
+        await self.mtcs.offset_camera_hexapod(x=0, y=0, z=z_move, u=0, v=0)
+        self.current_z_position = target_z  # Update current position
+        self.log.debug(f"Moved hexapod to z={target_z} (relative move: {z_move}).")
 
-        # Move the hexapod back to in focus position
-        await self.mtcs.offset_camera_hexapod(x=0, y=0, z=self.dz, u=0, v=0)
+    async def take_cwfs_image(self, reason_suffix: str) -> int:
```

Comment:
I don't see the value of having this method here. Why not just call `self.camera.take_cwfs` directly? note that, because this is a method, you need to add `self.supplemented_group_id` as a class attribute. I suggest you get rid of this method (and also remove `self.supplemented_group_id` in favor of a local variable. If you really want to have a method to encapsulate this operation (it does make the inline call leaner), pass `supplemented_group_id` alongside `reason_suffix` instead of relying on a class attribute.